### PR TITLE
Add BraveTab class and migrate chromium_src overrides to it.

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/24305

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/chrome/browser/ui/views/tabs/tab_strip.cc [hunk 1]
```diff
@@ -4,10 +4,21 @@
...
 
 #include "chrome/browser/ui/views/tabs/tab_strip.h"
+
+#include "brave/browser/ui/views/tabs/brave_tab.h"
 #include "brave/browser/ui/views/tabs/brave_tab_hover_card_controller.h"
+#include "chrome/browser/ui/views/tabs/browser_tab_strip_controller.h"
+#include "chrome/browser/ui/views/tabs/tab_container.h"
+
+#if BUILDFLAG(IS_WIN)
+#include "ui/gfx/win/hwnd_util.h"
```
Comment (bridiver) [inline chromium_src/chrome/browser/ui/views/tabs/tab_strip.cc:14]: why this include?

File: chromium_src/chrome/browser/ui/views/tabs/tab_strip.cc [hunk 2]
```diff
@@ -4,10 +4,21 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "chrome/browser/ui/views/tabs/tab_strip.h"
```
Comment (bridiver) [inline chromium_src/chrome/browser/ui/views/tabs/tab_strip.cc:6]: are all of these includes still relevant?

**Review by bridiver — APPROVED** [review]

File: chromium_src/chrome/browser/ui/views/tabs/tab_strip.cc (hunk 1 above)
Comment (fallaciousreasoning) [reply chromium_src/chrome/browser/ui/views/tabs/tab_strip.cc:14]: Without it the build was failing on Windows. Somewhere deep inside here there's a Windows `AddTab` function which was getting replaced by our overriding definition of `AddTab`. This fixes that by including the file first.

File: chromium_src/chrome/browser/ui/views/tabs/tab_strip.cc (hunk 2 above)
Comment (fallaciousreasoning) [reply chromium_src/chrome/browser/ui/views/tabs/tab_strip.cc]: Removed that one - the others all seem to be necessary.

**Pull request merged**